SYCL/ESIMD/api/functional/operators/operator_bitwise_not.cpp

```diff
@@ -0,0 +1,176 @@
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed = run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed = run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         allocator);
+    shared_vector<DataT> source_simd_result(NumElems, allocator);
+    shared_vector<DataT> operator_result(NumElems, allocator);
+
+    shared_element<bool> is_correct_type(queue, true);
+
+    queue.submit([&](sycl::handler &cgh) {
+      const DataT *const ref = shared_ref_data.data();
+      DataT *const source_simd_result_data_ptr = source_simd_result.data();
+      DataT *const operator_result_data_ptr = operator_result.data();
+      auto is_correct_type_ptr = is_correct_type.data();
+
+      cgh.single_task<Kernel<DataT, NumElems, TestCaseT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            *is_correct_type_ptr =
+                TestCaseT::template call_operator<DataT, NumElems>(
+                    ref, source_simd_result_data_ptr, operator_result_data_ptr);
+          });
+    });
+    queue.wait_and_throw();
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      if (!are_bitwise_equal(ref_data[i], source_simd_result[i])) {
+        passed = false;
+
+        const auto description = operators::TestDescription<DataT, NumElems>(
+            i, source_simd_result[i], ref_data[i], data_type);
+        log::fail(description);
+      }
+
+      DataT retrieved = operator_result[i];
+      DataT expected = ~shared_ref_data[i];
+      if (!are_bitwise_equal(expected, retrieved)) {
+        passed = false;
+        const auto description = operators::TestDescription<DataT, NumElems>(
+            i, retrieved, expected, data_type);
+        log::fail(description);
+      }
+    }
+
+    if (!is_correct_type.value()) {
+      passed = false;
+      log::note("Test failed due to type of the object that returns " +
+                TestCaseT::get_description() +
+                " operator is not equal to the expected one for simd<" +
+                data_type + ", " + std::to_string(NumElems) + ">.");
+    }
+
+    return passed;
+  }
+};
+
+int main(int, char **) {
+  sycl::queue queue(esimd_test::ESIMDSelector{},
+                    esimd_test::createExceptionHandler());
+
+  bool passed = true;
+
+  const auto uint_types = get_tested_types<tested_types::uint>();
```

Comment:
```suggestion
  // Running test for all integral types
  const auto uint_types = get_tested_types<tested_types::uint>();
```

Reply:
Same comment has been added [here](https://github.com/intel/llvm-test-suite/pull/850/commits/66e0d60d08be20c2069e90e626ef302a7446303a#diff-438d222b55125297a312d73be031ed90355f82de7eee316af778345a1c741a0dR43) and [here](https://github.com/intel/llvm-test-suite/pull/850/commits/66e0d60d08be20c2069e90e626ef302a7446303a#diff-c2cae8f61bdba7b88270c263569fede5cea9bb1fea2c721d60c10f9407b16f41R37)